coreclient/src/utils/data_migrations.rs

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2025 Phoenix R&D GmbH <hello@phnx.im>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! Data migrations implemented in Rust that cannot be expressed in SQL.
+
+use mimi_content::content_container::MimiContentV1;
+use phnxcommon::codec::{BlobDecoded, BlobEncoded, PhnxCodec};
+use sqlx::{SqlitePool, migrate::Migrate, query, query_as};
+use tokio_stream::StreamExt;
+use tracing::{error, info};
+
+use crate::{ConversationMessageId, conversations::messages::persistence::VersionedMessage};
+
+const MESSAGE_STATUS_MIGRATION_VERSION: i64 = 20250703133517;
+
+/// Migrate data in the database that cannot be expressed in SQL.
+pub(crate) async fn migrate(pool: &SqlitePool) -> Result<(), sqlx::Error> {
+    let Some(migrations) = pool.acquire().await?.list_applied_migrations().await.ok() else {
+        // The migrations might not yet exist
+        return Ok(());
+    };
+
+    let has_message_status = migrations
+        .iter()
+        .any(|m| m.version == MESSAGE_STATUS_MIGRATION_VERSION);
+    if !has_message_status {
+        if let Err(error) = convert_messages_v1_to_v2(pool).await {
+            error!(%error, "Failed to convert messages from version 1 to version 2");
+        }
+    }
+
+    Ok(())
+}
+
+/// Convert all versioned v1 messages to v2 messages in the database.
+async fn convert_messages_v1_to_v2(pool: &SqlitePool) -> anyhow::Result<()> {
```

Comment:
Is there an easy way to test this on CI? We don't have to do it for this PR, but we should at some point.

Reply:
I added a unit test.